ls1intum/Artemis
# Group courses by semesters in the managament overview

### Checklist

### Motivation and Context
Follow-up to #2215.

### Description
Groups the courses by semester. Courses without a semester will appear in the "Semester: unset" section. The newest semester will be sorted atop and uncollapsed, while the others are collapsed by default.

### Steps for Testing

1. Navigate to the courses overview
2. Check if all courses still appear and in the correct semester section

### Screenshots
The course overview with collapsed semester sections:
<img width="1597" src="https://user-images.githubusercontent.com/72132281/97847965-b3d8f100-1cf0-11eb-87a6-9f01ca91b74f.png">
When viewing several semester sections. Not that the tables do not align. This is expected and will be fixed in the future when the layout of the courses list changes.
<img width="1346" src="https://user-images.githubusercontent.com/72132281/97847970-b63b4b00-1cf0-11eb-8bd3-99936cf007ec.png">
If there are test courses, an extra section for them will appear (and they will not appear in their Semester section!):
![grafik](https://user-images.githubusercontent.com/72132281/97854371-a5db9e00-1cf9-11eb-929a-e109bd325e5d.png)
